flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/CepRuntimeContext.java

```diff
@@ -0,0 +1,196 @@
+import org.apache.flink.api.common.state.ReducingStateDescriptor;
+import org.apache.flink.api.common.state.ValueState;
+import org.apache.flink.api.common.state.ValueStateDescriptor;
+import org.apache.flink.metrics.MetricGroup;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A wrapper class for pattern select function and iterative condition function's {@link RuntimeContext}.
+ * The runtime context only supports basic operations. Consequently, state access, accumulators,
+ * broadcast variables and the distributed cache are disabled.
+ */
+public class CepRuntimeContext implements RuntimeContext {
+
+	private final RuntimeContext runtimeContext;
+
+	public CepRuntimeContext(RuntimeContext runtimeContext) {
+		this.runtimeContext = runtimeContext;
+	}
+
+	@Override
+	public String getTaskName() {
+		return runtimeContext.getTaskName();
+	}
+
+	@Override
+	public MetricGroup getMetricGroup() {
+		return runtimeContext.getMetricGroup();
+	}
+
+	@Override
+	public int getNumberOfParallelSubtasks() {
+		return runtimeContext.getNumberOfParallelSubtasks();
+	}
+
+	@Override
+	public int getMaxNumberOfParallelSubtasks() {
+		return runtimeContext.getMaxNumberOfParallelSubtasks();
+	}
+
+	@Override
+	public int getIndexOfThisSubtask() {
+		return runtimeContext.getIndexOfThisSubtask();
+	}
+
+	@Override
+	public int getAttemptNumber() {
+		return runtimeContext.getAttemptNumber();
+	}
+
+	@Override
+	public String getTaskNameWithSubtasks() {
+		return runtimeContext.getTaskNameWithSubtasks();
+	}
+
+	@Override
+	public ExecutionConfig getExecutionConfig() {
+		return runtimeContext.getExecutionConfig();
+	}
+
+	@Override
+	public ClassLoader getUserCodeClassLoader() {
+		return runtimeContext.getUserCodeClassLoader();
+	}
+
+	// -----------------------------------------------------------------------------------
+	// Unsupported operations
+	// -----------------------------------------------------------------------------------
+
+	@Override
+	public <V, A extends Serializable> void addAccumulator(
+		String name, Accumulator<V, A> accumulator) {
+		throw new UnsupportedOperationException("Accumulators are not supported.");
+	}
+
+	@Override
+	public <V, A extends Serializable> Accumulator<V, A> getAccumulator(String name) {
+		throw new UnsupportedOperationException("Accumulators are not supported.");
+	}
+
+	@Override
+	public Map<String, Accumulator<?, ?>> getAllAccumulators() {
+		throw new UnsupportedOperationException("Accumulators are not supported.");
+	}
+
+	@Override
+	public IntCounter getIntCounter(String name) {
+		throw new UnsupportedOperationException("Int counters are not supported.");
+	}
+
+	@Override
+	public LongCounter getLongCounter(String name) {
+		throw new UnsupportedOperationException("Long counters are not supported.");
+	}
+
+	@Override
+	public DoubleCounter getDoubleCounter(String name) {
+		throw new UnsupportedOperationException("Double counters are not supported.");
```

Comment:
I think getXXXCounter is very useful if user want using MetricGroup, So, I suggest exports getXXXCounter to user, What do you think?

Reply:
I don't think they have anything to do with metrics. I think they are rarely used nowadays and I would like to limit the feature set of the context as much as possible to decrease complexity of CEP.

Reply:
In our business applications, users will use this to statistically process data metrics, but I agree with the principle of minimal change.

Reply:
+1 to make the minimal changes at present and we can expose more functionalities in the future if needed.